src/framework/theme/components/select/option.component.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * @license
+ * Copyright Akveo. All Rights Reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  EventEmitter,
+  HostBinding,
+  HostListener,
+  Inject,
+  InjectionToken,
+  Input,
+  Output,
+} from '@angular/core';
+import { convertToBoolProperty } from '../helpers';
+
+
+/**
+ * Token that provides `NbSelectComponent` for child options.
+ * */
+export const NB_SELECT = new InjectionToken('select');
```

Comment:
can't we just inject NbSelectComponent?

Reply:
Sure, we can. This is idea how we can bypass circular dependencies.